hack/generated/controllers/generic_controller.go

```diff
@@ -426,32 +433,38 @@ func (gr *GenericReconciler) MonitorDeployment(ctx context.Context, action Recon
 		return ctrl.Result{}, err
 	}
 
+	deployment, err, retryAfter := gr.ARMClient.GetDeployment(ctx, deployment.Id)
+	if err != nil {
+		if retryAfter != 0 {
+			data.log.Info("Error GETting deployment, will retry", "delay", retryAfter/time.Second)
```

Comment:
need to log error as we are hiding it?

Reply:
Same comments as above -- `performing GET deployment` and `delaySec`